assets/js/autocomplete/autocomplete-list.js

```diff
@@ -89,12 +90,63 @@ export function moveAutocompleteSelection (offset) {
 
   if (elementToSelect) {
     elementToSelect.classList.add('selected')
+    showPreview(elementToSelect)
+    
     elementToSelect.scrollIntoView({ block: 'nearest' })
   } else {
     qs(AUTOCOMPLETE_SUGGESTION_LIST_SELECTOR).scrollTop = 0
   }
 }
 
+/**
+ * Toggles the preview state of the autocomplete list
+ */
+export function togglePreview () {
+  state.previewOpen = !state.previewOpen;
+  const suggestionList = qs(AUTOCOMPLETE_SUGGESTION_LIST_SELECTOR)
+  if (state.previewOpen) {
+    suggestionList.classList.add('previewing')
+    const elementToSelect = qs(`${AUTOCOMPLETE_SUGGESTION_SELECTOR}[data-index="${state.selectedIdx}"]`)
+    showPreview(elementToSelect)
+  } else {
+    suggestionList.classList.remove('previewing')
+    const container = previewContainer()
+
+    if (container) {
+      container.remove()
+    };
+  }
+}
+
+function showPreview(elementToSelect) {
+  const container = previewContainer()
+
+  if (container) {
+    container.remove()
+  };
+
+  const suggestionList = qs(AUTOCOMPLETE_SUGGESTION_LIST_SELECTOR)
+
+  if(state.previewOpen && elementToSelect) {
+    suggestionList.classList.add('previewing')
+    const newContainer = document.createElement('div')
+    newContainer.classList.add('autocomplete-preview')
+    const iframe = document.createElement('iframe')
+    const previewHref = elementToSelect.href.replace('.html', '.html?preview=true')
+    // The minimum permissions necessary for the iframe to run JavaScript and communicate with the parent window.
+    iframe.setAttribute('sandbox', 'allow-scripts allow-same-origin allow-popups allow-forms')
```

Comment:
Quick question: is there a reason why you add allow-popups and allow-forms?

Reply:
IIRC it was necessary for the iframe to render properly, but it was something I set early so we should explore removing it. I think `allow-popups` was to allow the hover info to work inside of previews.